rsa_client.cpp

```diff
@@ -111,12 +111,10 @@ openssl_unique_ptr<EVP_PKEY>	Rsa_client::load_private_key (uintptr_t key_id, RSA
 		throw Openssl_error(ERR_get_error());
 	}
 
-	rsa->n = BN_dup(public_rsa->n);
-	if (!rsa->n) {
-		throw Openssl_error(ERR_get_error());
-	}
-	rsa->e = BN_dup(public_rsa->e);
-	if (!rsa->e) {
+	const BIGNUM* n;
+	const BIGNUM* e;
```

Comment:
Please initialize `n` and `e` to `NULL`.  There are some code paths in `RSA_get0_key` which could leave `n` and `e` unmodified.

Reply:
I have changed the code to initialize all the `BIGNUM` pointers to `NULL`.